scripts/backfill_source_tags_for_unregistered_contributors.py

```diff
@@ -0,0 +1,181 @@
+from datetime import timedelta
+import logging
+from django.apps import apps
+from website.app import init_app
+from scripts import utils as script_utils
+import datetime
+import progressbar
+from django.core.paginator import Paginator
+
+import sys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def backfill_source_tags_for_osf4m_unregistered_contributors(dry_run):
+    """ Backfill osf4m source tags to all osf4m unregistered contributors
+    """
+    Tag = apps.get_model('osf', 'Tag')
+    OSFUser = apps.get_model('osf', 'OSFUser')
+    NodeLog = apps.get_model('osf', 'NodeLog')
+    meeting_source_tag = Tag.all_tags.get(name='source:campaign|osf4m', system=True)
+    meeting_nodes = meeting_source_tag.abstractnode_tagged.all()
+    logging.info('Number of meeting nodes found: ' + str(len(meeting_nodes)))
```

Comment:
I agree with the string concatenation. I did do a bit of research on `len()` vs `Queryset.count()`. It seems that `count()` does another SQL `count` query. If we're iterating through `meeting_nodes`, that means all of the contents are loaded into memory, and `len()` may have better performace over `count()`.

Reply:
I *think* (and I'm pretty new to QuerySet things, so grain of salt) that you'd need to store the queryset to a list first to get a benefit from using len(). Also, Postgres and Django will do their own caching things, so it might not be an actual penalty to do another sql command. Pushing to a list will cause additional memory pressure, though, which shouldn't happen with just using the queryset directly.

Reply:
According to Django [docs](https://docs.djangoproject.com/en/2.2/topics/db/queries/#caching-and-querysets)

> The first time a QuerySet is evaluated – and, hence, a database query happens – Django saves the query results in the QuerySet’s cache and returns the results that have been explicitly requested (e.g., the next element, if the QuerySet is being iterated over). Subsequent evaluations of the QuerySet reuse the cached results.

So I believe using count and len is roughly the same, as the QuerySet, when evaluated, are cached in memory. We do not have to cast the QuerySet into a list to get the benefit of using len().